assembly/src/parsers/u32_ops.rs

```diff
@@ -29,175 +29,123 @@ pub fn parse_u32assert(op: &Token) -> Result<Node, ParsingError> {
     }
 }
 
-/// Returns `U32CheckedAdd` instruction node if no immediate value is provided or
-/// `U32CheckedAddImm` instruction node otherwise.
-///
-/// # Errors
-/// Returns an error if the instruction token contains wrong number of parameters, or if the
-/// provided parameter is not a u32 value.
-pub fn parse_u32checked_add(op: &Token) -> Result<Node, ParsingError> {
-    debug_assert_eq!(op.parts()[0], "u32checked_add");
-    match op.num_parts() {
-        0 => unreachable!(),
-        1 => Ok(Instruction(U32CheckedAdd)),
-        2 => {
-            let value = parse_param::<u32>(op, 1)?;
-            Ok(Instruction(U32CheckedAddImm(value)))
+macro_rules! parseop {
+    ($(#[$attr:meta])* fn $fn: ident $op: literal => ( $ins: ident , $imm_ins: ident)) => {
+        $(#[$attr])*
+        pub fn $fn(op: &Token) -> Result<Node, ParsingError> {
+            debug_assert_eq!(op.parts()[0], $op);
+            match op.num_parts() {
+                0 => unreachable!(),
+                1 => Ok(Instruction($ins)),
+                2 => {
+                    let value = parse_param::<u32>(op, 1)?;
```

Comment:
This is my motivation for the macro, the line reduction is a nice to have, but what I really wanted was an easy way to see that all parse functions were using `u32` data type.